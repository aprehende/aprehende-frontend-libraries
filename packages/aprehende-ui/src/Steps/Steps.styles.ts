import styled from '@emotion/styled';

export const Container = styled.div`
  display: flex;
`;

export const StepItem = styled.div<{
  done?: boolean;
  current?: boolean;
}>`
  display: flex;
  gap: clamp(0.25rem, 2vw, 0.5rem);
  align-items: center;
  &:before {
    content: '';
    display: block;
    width: ${({ done }) => (done ? '0' : 'clamp(1.5rem, 5vw, 3rem);')};
    height: ${({ done }) => (done ? '0' : 'clamp(1.5rem, 5vw, 3rem);')};
    border-radius: 50%;
    border: ${({ done, current }) =>
      done ? 'none' : current ? '2px solid #e0e0e0' : '1px solid #e0e0e0'};
  }

  & > svg {
    width: clamp(1.5rem, 5vw, 3rem);
    height: clamp(1.5rem, 5vw, 3rem);
  }

  &:not(:last-child) {
    flex: 1;

    &:after {
      content: '';
      position: relative;
      z-index: -1;
      height: 2px;
      background-color: #e0e0e0;
      flex: 1;
      margin-right: 0.5rem;
    }
  }
`;

export const StepTitle = styled.h3`
  font-weight: bold;
  font-size: clamp(1rem, 4vw, 1.25rem);
`;
