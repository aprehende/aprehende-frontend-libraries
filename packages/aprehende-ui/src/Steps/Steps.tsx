import { BsFillCheckCircleFill } from 'react-icons/bs';

import { Container, StepItem, StepTitle } from './Steps.styles';
interface StepItem {
  title: string;
}

interface StepsProps {
  current: number;
  items: Array<StepItem>;
}

const Steps: React.FC<StepsProps> = ({ items, current }) => {
  return (
    <Container>
      {items.map((item, index) => (
        <StepItem
          key={index}
          done={index + 1 < current}
          current={current === index + 1}
        >
          {index + 1 < current && <BsFillCheckCircleFill color="#0984e3" />}
          <StepTitle>{item.title}</StepTitle>
        </StepItem>
      ))}
    </Container>
  );
};

export default Steps;
