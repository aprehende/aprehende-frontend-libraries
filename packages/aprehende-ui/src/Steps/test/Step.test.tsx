import { render } from '@testing-library/react';
import { describe, it } from 'vitest';

import Steps from '../Steps';

const current = 2;
const steps = [{ title: 'Paso 1' }, { title: 'Paso 2' }, { title: 'Paso 3' }];

describe('Steps', () => {
  it('should render correctly', () => {
    const { container } = render(<Steps current={current} items={steps} />);
    expect(container).toMatchSnapshot();
  });

  it('should render icon when step is done', () => {
    const { getByTestId } = render(<Steps current={current} items={steps} />);
    expect(getByTestId('step-icon')).toBeInTheDocument();
  });

  it('should not render icon when step is not done', () => {
    const { queryByTestId } = render(<Steps current={1} items={steps} />);
    expect(queryByTestId('step-icon')).not.toBeInTheDocument();
  });
});
