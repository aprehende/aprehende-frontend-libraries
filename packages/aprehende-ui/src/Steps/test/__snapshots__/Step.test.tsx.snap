// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Steps > should render correctly 1`] = `
<div>
  <div
    class="css-ov1ktg"
  >
    <div
      class="css-tqghp"
    >
      <svg
        color="#0984e3"
        data-testid="step-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(9, 132, 227);"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-3.97-3.03a.75.75 0 0 0-1.08.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-.01-1.05z"
        />
      </svg>
      <h3
        class="css-1jzba66"
      >
        Paso 1
      </h3>
    </div>
    <div
      class="css-16i4vku"
    >
      <h3
        class="css-1jzba66"
      >
        Paso 2
      </h3>
    </div>
    <div
      class="css-bfv55k"
    >
      <h3
        class="css-1jzba66"
      >
        Paso 3
      </h3>
    </div>
  </div>
</div>
`;
